sdk/src/trace/batch_span_processor.cc

```diff
@@ -180,11 +180,16 @@ void BatchSpanProcessor::DrainQueue()
 
 bool BatchSpanProcessor::Shutdown(std::chrono::microseconds timeout) noexcept
 {
-  is_shutdown_.store(true);
+  bool already_shutdown = is_shutdown_.exchange(true);
 
-  cv_.notify_one();
-  worker_thread_.join();
-  if (exporter_ != nullptr)
+  if (worker_thread_.joinable())
+  {
+    cv_.notify_one();
+    worker_thread_.join();
+  }
+
+  // Should only shutdown exporter ONCE.
+  if (!already_shutdown && exporter_ != nullptr)
```

Comment:
Is showing down is going, later call to `Shutdown` will just return true even if the previous `Shutdown` has not been completed?

Reply:
Yes and thanks, it's definitely a problem.I add a mutex lock to prevent this.